src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -219,13 +276,20 @@ def __fetch_randomly_sampled_root_transactions(
         )
         builder.add_conditions([Condition(lhs=Column("modulo_num"), op=Op.EQ, rhs=0)])
         snuba_query = builder.get_snql_query().query
-        groupby = snuba_query.groupby + [Column("modulo_num")]
+
+        extra_select = [
+            Function("has", [Column("contexts.key"), TRACE_PARENT_SPAN_CONTEXT], alias="is_root")
+        ]
+        snuba_query = snuba_query.set_select(snuba_query.select + extra_select)
+
+        groupby = snuba_query.groupby + [Column("modulo_num"), Column("contexts.key")]
         snuba_query = snuba_query.set_groupby(groupby)
```

Comment:
Nitpick (feel free to ignore)
```suggestion
        snuba_query = snuba_query.set_groupby(snuba_query.groupby + [Column("modulo_num"), Column("contexts.key")])
```